packages/components/src/popover/index.js

```diff
@@ -485,97 +480,38 @@ const Popover = (
 		__unstableBoundaryParent,
 	] );
 
-	// Event handlers for closing the popover.
-	const closeEventRef = useRefEffect(
-		( node ) => {
-			function maybeClose( event ) {
-				// Close on escape.
-				if ( event.keyCode === ESCAPE && onClose ) {
-					event.stopPropagation();
-					onClose();
-				}
-			}
-
-			node.addEventListener( 'keydown', maybeClose );
-
-			return () => {
-				node.removeEventListener( 'keydown', maybeClose );
-			};
-		},
-		[ onClose ]
-	);
-
-	const constrainedTabbingRef = useConstrainedTabbing();
-	const focusReturnRef = useFocusReturn();
-	const focusOnMountRef = useFocusOnMount( focusOnMount );
-	const focusOutsideProps = useFocusOutside( handleOnFocusOutside );
-	const mergedRefs = useMergeRefs( [
-		ref,
-		containerRef,
-		// Don't register the event at all if there's no onClose callback.
-		onClose ? closeEventRef : null,
-		focusOnMount ? constrainedTabbingRef : null,
-		focusOnMount ? focusReturnRef : null,
-		focusOnMount ? focusOnMountRef : null,
-	] );
-
-	/**
-	 * Shims an onFocusOutside callback to be compatible with a deprecated
-	 * onClickOutside prop function, if provided.
-	 *
-	 * @param {FocusEvent} event Focus event from onFocusOutside.
-	 */
-	function handleOnFocusOutside( event ) {
-		// Defer to given `onFocusOutside` if specified. Call `onClose` only if
-		// both `onFocusOutside` and `onClickOutside` are unspecified. Doing so
-		// assures backwards-compatibility for prior `onClickOutside` default.
-		if ( onFocusOutside ) {
+	const onDialogClose = ( type, event ) => {
+		// Ideally the popover should have just a single onClose prop and
+		// not three props that potentially do the same thing.
+		if ( type === 'focus-outside' && onFocusOutside ) {
 			onFocusOutside( event );
```

Comment:
Should we deprecate this in the future as well, in favour of onClose?

Reply:
Sounds like something we could try, I'm not certain yet of the impact.